objects/drawable.c

```diff
@@ -225,6 +233,7 @@ drawable_class_setup(lua_State *L)
     static const struct luaL_Reg drawable_meta[] =
     {
         LUA_OBJECT_META(drawable)
+        { "__gc", luaA_collect },
```

Comment:
I don't think this works. `common/luaclass.c`, `luaA_class_setup()` sets up its own `__gc` handler, so only one of the two will be executed (I'm not sure which one wins; if your "collect" signal is being emitted, then apparently your new code won). Instead, you want to add this `emit_signal` to `drawable_wipe()` which is called by the other `__gc`-handler.